### ISSUE TITLE
replace transparent proofs with zk proofs

### ISSUE THREAD

**Issue by redshiftzero** [penumbra-zone/penumbra#450]
Per #2 the current plan is to use PLONK. Next steps:

- [x] survey PLONK implementations
- [ ] replace [transparent Spend and Output proofs](https://github.com/penumbra-zone/penumbra/blob/main/crypto/src/proofs/transparent.rs) with zk proofs
- [x] lift proofs from the [spend- or output- level](https://github.com/penumbra-zone/penumbra/blob/main/proto/proto/transaction.proto#L60) to the transaction level - meaning a single proof for the N spend/outputs in the transaction - in https://github.com/penumbra-zone/penumbra/pull/489

**Comment by hdevalence** [comment]
We might want to do the third point first, and separately from any of the actual proof implementation, in order to "pre-align" our code with what we'll want to do eventually.  This involves figuring out how we want to derive the proof statements from the transaction actions, before we actually go and implement them.

**Comment by redshiftzero** [comment]
Of the PLONK implementations out there, [`ark-plonk`](https://github.com/ZK-Garage/plonk) and [`jf-plonk`](https://github.com/EspressoSystems/jellyfish) both use arkworks under the hood, as we do in `decaf377`. We've tried out both libraries (see #665 for jf-plonk and and #666 for ark-plonk) for implementing part of the output proof.

Benchmarks on those branches are as follows:
* preprocessing time is 10.6s for `ark-plonk` vs 5s for `jf-plonk` (so still good that the proof lifting was removed for now as explained in #528)
* proving speed is 10.8s for `ark-plonk` vs 7.8s for `jf-plonk`
* verification speed is 58ms for `ark-plonk` vs 25ms for `jf-plonk`

In terms of dev experience, my take is that `ark-plonk` was very straightforward to use, I ran into some issues with `jf-plonk` e.g. the `WrongQuotientPolyDegree` mentioned in the diff in #665 (probably my error to be fair).

One other point worth noting is the `jf-plonk` library uses [Merlin](https://merlin.cool/) and so wasn't vulnerable to the Fiat-Shamir transformation issues described [here](https://blog.trailofbits.com/2022/04/13/part-1-coordinated-disclosure-of-vulnerabilities-affecting-girault-bulletproofs-and-plonk/).

tl;dr - `jf-plonk` faster by 2x in verification and circuit preprocessing time (relevant if we want to eventually lift the proofs to the tx level in the future), `ark-plonk` a little smoother to use dev wise
